addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/pscan/WebSocketPassiveScanThread.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.websocket.pscan;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.db.DatabaseException;
+import org.zaproxy.zap.extension.websocket.alerts.WebSocketAlertWrapper;
+import org.zaproxy.zap.extension.websocket.WebSocketMessage;
+import org.zaproxy.zap.extension.websocket.WebSocketMessageDTO;
+import org.zaproxy.zap.extension.websocket.WebSocketProxy;
+import org.zaproxy.zap.extension.websocket.WebSocketSenderListener;
+import org.zaproxy.zap.extension.websocket.db.TableWebSocket;
+
+import java.util.Iterator;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * Implements a background thread for passive scanning
+ */
+public class WebSocketPassiveScanThread extends Thread implements WebSocketSenderListener {
+    
+    private static final Logger LOGGER = Logger.getLogger(WebSocketPassiveScanThread.class);
+    
+    public static final int WEBSOCKET_LISTENER_ORDER = 20; //TODO Make it more formal
```

Comment:
Does something here actually need to change or is this TODO out of date?

Reply:
tbh I am not sure If there is any "official" order with which listeners should be arranged. So 20 just means that it has not to be informed prior to any other listener.

Reply:
The order doesn't really matter here, right? It's just to know which messages to scan, this could be private in that case.

The thread could be a `WebSocketObserver` and be after `WebSocketStorage`, that would ensure the messages are already persisted (we would miss the sender though...).